### ISSUE TITLE
Duets are broken

### ISSUE THREAD

**Issue by AlecScudder** [Vocaluxe/Vocaluxe#370]
Just tried the latest build. I like the new look (I'm using very old build), but it appears duets no longer work.

P1 lines
-Lyrics display for both P1 and P2 (bad)
-Notes display for both P1 and P2 (bad)

P2 lines
-Lyrics display for P2 only (good)
-Notes do not display at all (bad)

**Comment by lukeIam** [comment]
I think I found the reason - the `VoiceNr` of the players get reseted.

We load the voice number with:
https://github.com/Vocaluxe/Vocaluxe/blob/c888b879d0a94ffe073c383eb2cec4182349efa9/Vocaluxe/Screens/CScreenNames.cs#L630-L638

But when we are faded to to SingScreen we are resetting it with 
```
_FinishScreenFading()->OnShowFinish()->_NextSong()->ResetPlayer()
```

I thing this code should save it (but it is called **after** the first reset and is followed be a second player rest):
https://github.com/Vocaluxe/Vocaluxe/blob/be9e54d7d5ece0311e902c41bee00ef4e1e54774/Vocaluxe/Screens/CScreenSing.cs#L1065-L1071

@flokuep @Jiiniasu 
Maybe this is a side effect to the big multi screen change.
Do you know the right order of methods that should be called to change to the sing screen?

**Comment by Jiiniasu** [comment]
@lukeIam Been testing around a bit and it seems the bug appeared in https://github.com/Vocaluxe/Vocaluxe/commit/52becec3d23472698fc94be8d015fed8155e4879 

Haven't had the time to dig into the code though.

**Comment by flokuep** [comment]
Yeah, the main problem is, that Game was reset.  I think the best place for fixing this is CGame directly. We might move game-related stuff to this one, e.g. have functions for starting next round, restarting round and restarting game. Currently it's a (crude) mix of CGame.Reset() CGame.ResetPlayer() and CGame.NextRound()

**Comment by AlecScudder** [comment]
Any plans to fix this? Would love to try some of the new features but I make all my songs into duets so the newer versions of the game are unplayable to me.

**Comment by diver66** [comment]
Hi Alex,
how may have you changed and which ones ? I am replacing my songs with better video versions. maybe we can exchange some.
regards
Joe

**Comment by AlecScudder** [comment]
diver66,

If there is a way I can private message you we can discuss.

**Comment by diver66** [comment]
Hi Alec,
my email is joe@la-palma-tauchen.com
I will delete this comment as soon as I receive a mail from you.

**Comment by lukeIam** [comment]
@diver66 github always saves the history of you posts -> I've deleted those posts

**Comment by lukeIam** [comment]
> Any plans to fix this?

I would like to, but I have not enough knowledge about this part of Vocaluxe to do so.
(And atm we don't have much time to use for Vocaluxe developing...)

**Comment by Jiiniasu** [comment]
Had a swing at this and made a fix that works (afaik). The code is butt-ugly worthy of crying blood, but that's what you get when I put in relatively low effort and time. I'd rather have something proper, but who knows how long that would take me.

But for those who don't want to dive into the code and just try things out: https://github.com/Jiiniasu/Vocaluxe/tree/horrible_duet_fix

**Comment by flokuep** [comment]
I guess the whole Screen-code is horrible, so from my side it would be okay to add this ;)
